server/src/main/java/org/elasticsearch/index/codec/vectors/IVFVectorsWriter.java

```diff
@@ -325,9 +342,12 @@ private void mergeOneFieldIVF(FieldInfo fieldInfo, MergeState mergeState) throws
 
                     CentroidSupplier centroidSupplier = createCentroidSupplier(
```

Comment:
It took me a while to understand this, I think we should not reuse the written centroids now that we have a complex structure but write them in the temp file in order.